server/datastore/mysql/migrations/tables/20230926082152_AddWindowsMDMTables.go

```diff
@@ -0,0 +1,70 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230926082152, Down_20230926082152)
+}
+
+func Up_20230926082152(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE windows_mdm_pending_commands (
+    -- command uuid is fleet-generated, so that a command has a unique id as
+    -- soon as it is enqueued. Note that a UUID is 36 chars, but for
+    -- consistency with nano_mdm_commands we use 127 here.
+    command_uuid    VARCHAR(127) NOT NULL,
+    device_id       VARCHAR(255) NOT NULL,
+
+    -- verb is e.g. Add, Exec, Get, etc.
+    cmd_verb        VARCHAR(20)  NOT NULL,
+    setting_uri     VARCHAR(255) NOT NULL,
+    setting_value   VARCHAR(255) NOT NULL,
+    data_type       VARCHAR(255) NOT NULL,
+    system_origin   TINYINT(1) NOT NULL DEFAULT FALSE,
+
+    created_at      TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at      TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+    PRIMARY KEY (command_uuid, device_id)
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create windows_mdm_commands table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE windows_mdm_commands (
```

Comment:
So I'm wondering, do we need two tables or could the `pending` one be represented by having a subset of fields set?

Alternatively, if we feel it's better to have both tables, could one refer to the other and not repeat all fields?

Just thinking out loud trying to see if there's a more efficient way to model this (while remaining easy to use in code of course), as there's quite a bit of overlap and they have the exact same PK.

Reply:
The `windows_mdm_pending_commands` is mostly to receive commands from fleet server/fleetctl

On the other hand, the `windows_mdm_commands ` table can contain commands that were not part of the `windows_mdm_pending_commands `table and were generated by the Fleet server in response to an incoming MDM command. That was the main idea of keeping two tables.